core/secret_scrub.go

```diff
@@ -0,0 +1,10 @@
+package core
+
+// ToScrubInfo stores the info to access secrets and scrub them from outputs.
+type SecretToScrubInfo struct {
+	// Envs stores environment variable names that we need to scrub.
+	Envs []string
+
+	// Files stores secret file paths that we need to scrub.
+	Files []string
```

Comment:
While not 100% necessary I still like to explicitly include JSON tags for anything we expect to marshal to/from JSON.

```suggestion
	Envs []string `json:"envs,omitempty"`

	// Files stores secret file paths that we need to scrub.
	Files []string `json:"files,omitempty"`
```